Week 1 new

travacry opened the pull request.

travacry committed:
change desc

travacry committed:
add ignore vendor directory

travacry committed:
add workflows

travacry committed:
- ignore up

travacry committed:
up 1.22

travacry committed:
- pkg chat_v1 updated

travacry committed:
- updated tree dirs

travacry committed:
- del main

travacry committed:
- go mod up

travacry committed:
- fix pkg

travacry committed:
- fix module

travacry committed:
- create cmd by server and client

travacry committed:
- fix check lint

travacry committed:
- fix deprecated Dial

travacry committed:
- fix lint local to 1.59 --fix

travacry committed:
- only panic + log errors

travacry committed:
- fix fatal

travacry committed:
- добавил методы и реф

travacry committed:
- then uml

travacry committed:
- added uml-seq diagram for understanding

travacry committed:
- added new methods
- corrections based on the first review

travacry committed:
- sort

travacry committed:
- fix printf

travacry committed:
- printf ++

travacry committed:
- err blocks

travacry committed:
- united blocks

travacry committed:
- fix nr, Id -> ID

travacry committed:
- fix nr

travacry committed:
- fix nr

travacry committed:
- fix Id->ID

travacry committed:
- del cID

travacry committed:
- del uID

travacry committed:
- combined error block and code block

travacry committed:
- main up

travacry committed:
- defer as a single block

travacry committed:
- sort imports

travacry committed:
- up go.mod go.sum

Shemistan commented on api/chat_v1/chat.proto:
```diff
@@ -0,0 +1,99 @@
+option go_package = "github.com/travacry/chat-server/pkg/chat_v1;chat_v1";
+
+service ChatV1 {
+  rpc Create(CreateRequest) returns (CreateResponse);
```
Хорошо бы уточнить что именно создается, удаляется, отправляется и т.д. Методов много и это может запутывать

travacry replied:
всеже обощение ItemState разделил на пару
enum UserState {
  USER_NOT_CONFIRM = 0; // user must connect
  USER_ACTIVE = 1;
  USER_BAN = 2;
  USER_DELETE = 3;
}

enum ChatState {
  CHAT_ACTIVE = 0;
  CHAT_DELETE = 1;
}

Connect - оставил как есть.

перегруппировал по коду.

Shemistan commented on api/chat_v1/chat.proto:
```diff
@@ -0,0 +1,99 @@
+  rpc ListUsers(ListUsersRequest) returns (ListUsersResponse);
+}
+
+enum ItemState {
```
согласно общим соглашениям о стиле в Protobuf, имена значений перечислений в enum обычно пишутся в верхнем регистре

Shemistan commented on uml/sequence.puml:
```diff
@@ -0,0 +1,125 @@
+@startuml
```
Прикольно!

travacry committed:
- clarified the names of the methods

travacry committed:
- regroup by context
- split ItemState into UserState and ChatState
- now functions have more specific names
